bam_to_mate_hist.py

```diff
@@ -603,9 +601,11 @@ def write_stat_table(stat_dict, outfile_name):
 
     stat_dict, totals, non_dups = parse_bam_file(num_reads=num_reads, bamfile=bamfile,
                                count_diff_refname_stub=count_diff_refname_stub)
-    observed_dup_rate, extrapolated_dup_rate, target_x = plot_dup_saturation(outfile_name, totals, non_dups, target_x=c_args['target_read_total'])
+    observed_dup_rate, extrapolated_dup_rate, target_x, V, K = plot_dup_saturation(outfile_name, totals, non_dups, target_x=c_args['target_read_total'])
```

Comment:
the bug that you observed with low read nums is i think related to a change that I made such that V and K are just not estimated if there are too few reads, due to failed convergence throwing an error with small numbers of reads. see above `plot_dup_saturation()` definition, 100K reads is the minimum number for which it will try. probably that could be handled in a better way.

Reply:
yes, i think that is a good idea. and a good point about single return statements, i rely on such tricks too much.